spec/controllers/image_sets_controller_spec.rb

```diff
@@ -88,4 +88,18 @@
         change { image_set.reload.main_image }.from(nil).to(main_image)
     }
   end
+
+  describe '#destroy' do
+    let(:image_set) { Fabricate :image_set_with_images }
+    let(:request) { -> { delete :destroy, id: image_set.id } }
+
+    it_behaves_like "an authenticated controller"
+    it { expect { subject }.to change { ImageSet.count }.by(1) }
+    it { expect { subject }.to change { Image.count }.by(5) }
+
```

Comment:
also you will need an exclamation mark in the image_set definition, `let!(:image_set)`  This means the objects gets created right away before the test starts. You need this since Rspec will be checking ImageSet.count before and after firing the request, so you need the image_set created first, then hopefully the controller deletes it and the count changes by -1.